File: api/routers/tasks.py
from __future__ import annotations

from fastapi import APIRouter, Depends, Request

from api.auth import verify_api_key
from shared.protocol import Task


router = APIRouter(
    prefix="/tasks", tags=["tasks"], dependencies=[Depends(verify_api_key)]
)


@router.get("/{job_id}", response_model=list[Task])
async def list_tasks(request: Request, job_id: str):
    coordinator = request.app.state.state.coordinator
    return await coordinator.list_tasks(job_id)

File: packages/api/src/routes/tasks.ts
import type { FastifyInstance } from "fastify";
import { verifyApiKey } from "../middleware/auth.js";
import type { AppState } from "../services/state.js";

export async function taskRoutes(app: FastifyInstance): Promise<void> {
  app.addHook("preHandler", verifyApiKey);

  app.get<{ Params: { job_id: string } }>(
    "/tasks/:job_id",
    async (req) => {
      const state = (app as unknown as { state: AppState }).state;
      return state.coordinator.listTasks(req.params.job_id);
    },
  );
}
